### ISSUE TITLE
[Feature Request] Blur options for transparencies in the UI

### ISSUE THREAD

**Issue by SyrisBit** [pop-os/cosmic-epoch#604]
I'd be nice to have some blur options for the UI elements that can be made to be transparent, or opacity adjusted, like the dock, panel, terminal window, etc., so that the transparent elements could be made to resemble frosted glass.

**Comment by Department-stack** [comment]
i believe they did say something about adding a "frost" effect in

**Comment by ahmadmuqri0** [comment]
Yeah would agree with this one, since terminal like alacritty support blur options, kinda nice to have one

**Comment by sleeplessKomodo** [comment]
if this is accomplished, I'm fully switching from KDE to Cosmic.

**Comment by juehang** [comment]
Any ETA or idea on how far down the roadmap this is?

**Comment by zoroaster1x** [comment]
> Any ETA or idea on how far down the roadmap this is?

https://old.reddit.com/r/pop_os/comments/1jcww0a/cosmic_with_blur_animations_folders_that_take_on/mi5wrz1/
>PopOS System76 Desktop Engineer mmstick 9 months ago
>Blur is a post-release feature

I guess a few updates after a lot of the common bugs are fleshed out and necessary desktop features are implemented. Also users expressing their desire for such a feature will likely also impact the time it takes to be added.

Hopefully they will implement it in a decent way. I quite like this for KDE:
https://github.com/4v3ngR/kwin-effects-glass

I use this with Zen Browser (On Zen I installed the transparency mod in the Zen Addons marketplace with the Zen Internet extension from Firefox store and then in Kwin desktop effects settings I enabled blur for all transparent windows and edited some config stuff) 

I wonder how Cosmic will implement blurring. I would like a frosted glass feature as well as some sort of easy API to have eg an app I develop request transparency from Cosmic to certain transparent elements, as well as partial transparency for certain regions of GUI apps for slick modern transparency. I already prefer Cosmic to Gnome. Blur my shell for Gnome is atrocious, ugly, and buggy in my opinion. I hope they don't do what was done there and do it in an efficient way.

I mentioned partial blur support for certain elements, it should be allowed to make applications blurred like this:
<img width="761" height="733" alt="Image" src="https://github.com/user-attachments/assets/d7edf5b4-a4ed-4e99-8457-1a3d77d3087e" />

**Comment by sethCodesSometimes** [comment]
The refraction/liquid glass effect in the kwin-effects-glass thing you linked looked pretty cool. Could such a thing be implemented in cosmic? It seems a little more difficult than just a blur.

**Comment by zoroaster1x** [comment]
> The refraction/liquid glass effect in the kwin-effects-glass thing you linked looked pretty cool. Could such a thing be implemented in cosmic? It seems a little more difficult than just a blur.

I don't see why not, though I'd imagine it to be low priority until bugs are fixed. Of course, you also gotta make it known to the devs it's something the users want.

<img width="1366" height="768" alt="Image" src="https://github.com/user-attachments/assets/2da6e727-99c4-4bb3-a312-14654d2d8d8a" />
The kwin-effects-glass thing is really cool and I hope cosmic implements something to allow this.

**Comment by zoroaster1x** [comment]
Seems this issue can be tracked here:

https://github.com/pop-os/cosmic-comp/issues/511
